### PR TITLE
ISPN-10867 AsyncStoreWithoutEvictionFunctionalTestsegmented creates d…

### PR DESCRIPTION
**Pull request by wburns**
…ata directories in core directory

https://issues.jboss.org/browse/ISPN-10867

### CODE REVIEW

File: core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java [hunk 1]
```diff
@@ -36,15 +38,10 @@
...
-
-   @BeforeClass
-   protected void setUpTempDir() {
-      tmpDirectory = TestingUtil.tmpDirectory(this.getClass());
-   }
+   private File tmpDirectory;
 
    @AfterClass
-   protected void clearTempDir() {
+   public void clearTempDir() {
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java]: Need `@AfterClass(alwaysRun = true)` as well to run after failures

Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java]: Sure

Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/support/AsyncStoreWithoutEvictionFunctionalTest.java]: Updated

**Comment by ryanemerson** [comment]
Thanks @wburns

**Pull request merged**